packages/process/src/node/process.ts

```diff
@@ -118,6 +120,10 @@ export abstract class Process {
         this.initialCwd = options && options.options && 'cwd' in options.options && options.options['cwd'].toString() || __dirname;
     }
 
+    public initClientConnection(connection: MessageConnection): void {
```

Comment:
Absolutely, I agree with you.  In fact I tried to move it out before submitting the PR but there was no obvious way without a lot more refactoring.  See my original comment in which I specifically mention the `initClientConnection` function.

The problem is that hooking up the connection is different in a process than in a terminal.  As the code was before, the hookup of the process with the connection was done in terminal-backend-contribution.  I moved this to 'process' package, renamed to process-backend-contribution, and moved the hookup implementation into Process.  ProcessManager only gives us the process, so we need to determine, given only the Process, how to do the hookup.  As long as all RawProcess processes have the same hookup, we are ok.  However I was also considering custom processes that may need different hookup.  For example if a task had to implement progress percentage on the backend then a message type would need to be added to report the percentage.  This means different hookups for different RawProcess processes.  This can be done by setting the initClientConnection function in the custom task runner after the process has been created.  This is not clean, but, as I said in the opening post, I believe the real solution would be to change ProcessManager so we get more than just the Process.  If we had the task type as well then we could determine the needed hookup from that.  I guess really I am waiting to see if changing ProgressManager in that way is possible or a no-go. 

It is true that I use `readline` for RawProcess but not TerminalProcess, and this does change the hookup. However there are other changes in the hookup too, so even if we passed data back line-by-line for TerminalProcess too, we would still need to different hookup.

> If anything it should either be somewhere else or not tied to a connection.  

I don't understand what you mean by 'not tied to a connection'.  The whole point of this is to connect a process to a connection.

Reply:
Sorry for the long replies.  I did not say that the place I really wanted to put initClientConnection was in the TaskRunner interface.  We just need to be able to get the task type from the process id so we can access the implementation of this interface when hooking up.

Reply:
> I don't understand what you mean by 'not tied to a connection'. The whole point of this is to connect a process to a connection.

I meant to find a way to abstract it more than directly hooking to a connection object, but I think we should just move this code elsewhere.

`RawProcess` and `TerminalProcess` represent conventionally spawned/forked process and pseudo-terminal/terminal launched processes, respectively. Since they represent something low/simple, maybe the `initConnection` could be moved to a wrapper for these processes (one abstraction level above, for the tasks' sake).

I think the way you did it is good, just a matter of organizing it elsewhere.

> I believe the real solution would be to change ProcessManager so we get more than just the Process.

Adding `TaskProcess` to wrap regular `Process` instances?